repo2docker/buildpacks/r.py

```diff
@@ -74,7 +74,8 @@ def r_version(self):
         }
 
         # the default if nothing is specified
-        r_version = "4.1"
+        # Use full version is needed here, so it a valid semver
+        r_version = version_map["4.1"]
```

Comment:
Hmm actually, looking at this code, it seems that r_version is pointless to set here. Whenever it used referenced later in this function, it is always set explicitly first independent of its previous value.

I think this assignment is irrelevant, and we missed that because its name is too similar to `_r_version`.

Reply:
Hmm I don't understand what you mean yet Yuvi

What I see is the function scoped new variable `r_version` is set on line 78. To me it seems that the value assigned is never read from the variable until the variable scope expires, making the assignment irrelevant.

Can you pinpoint the location in the code the assigned value on line 78 is influencing something?

Reply:
@consideRatio read in line 92 here: https://github.com/jupyterhub/repo2docker/pull/1161/files#diff-3fa4ebd097693f90be6a3f6fce113df19b43c5c3efcb9c5b0070432c09249b2fR92

Reply:
you'll note that it's outside the if defined in line 84, so if that if branch is not taken, this value is read and used.

Reply:
Ah sorry for the noise!

Reply:
No worries, @consideRatio! I hope the comment helps clarify it!